REST API: Services API section

mariodavid opened the pull request.

knstvk commented on content/modules/rest/pages/business-logic.adoc:
```diff
@@ -1,20 +1,208 @@
 = Business Logic
 
-== Service Invocation
+When interacting with the REST API it is oftentimes needed to have an application-level layer of business logic in place that represents the Invocation point for the API. It can be used for orchestration, validation, or other tasks that should happen when the API Client interacts with the system. The xref:rest:entities-api.adoc[Entities API] does not allow placing additional orchestration busines ...
```
"the Invocation point, the API Client" - we usually don't use capitalization if it's not a proper noun or an object in code.
There are other cases below: "Service, Service Interface". 
Plz remove capitalization in these words, it's better for uniformity and IMHO for readability too.

mariodavid replied:
done

knstvk commented on content/modules/rest/pages/business-logic.adoc:
```diff
@@ -1,20 +1,208 @@
+
+To fulfill the first requirement of implementing a Java interface, let's look at the following example:
+
+The Service Interface allows to expose the methods via the REST Services API:
```
"The service interface allows you to expose ..."

mariodavid replied:
done

knstvk commented on content/modules/rest/pages/business-logic.adoc:
```diff
@@ -1,20 +1,208 @@
+----
+public interface CalculationService {
+
+    String NAME = "sample_CalculationService"; // <1>
```
We actually don't use constants for bean names in the framework and Studio doesn't generate them in the apps.
So this part should be rewritten.

mariodavid replied:
done

knstvk commented on content/modules/rest/pages/business-logic.adoc:
```diff
@@ -1,20 +1,208 @@
+jmix.rest.servicesConfig = rest/sample/rest-services.xml
+----
+
+Once the Service is exposed through the Services API, it is possible to invoke the service as an API Client.
```
probably "it is possible to invoke the service **from** an API client."

mariodavid replied:
done

knstvk commented on content/modules/rest/pages/business-logic.adoc:
```diff
@@ -1,20 +1,208 @@
+
+* Entities
+* Entity Collections
+* serializable POJOs
```
Serializable POJOs

mariodavid replied:
done

knstvk reviewed (changes requested):
General note: what about taking examples from the real code of the app?

mariodavid commented:
@knstvk I made the following changes/additions:

* switched examples to be aligned with the order management example
* added custom controller section
* Added information about `@Service` requirement instead of the interface

I would propose to merge this section if there are no further problems with the content as is. I guess this section can be improved with the information about datatypes and the security behavior of the custom controller if this is under change right now. But I would do this then later.
